codegen/smithy-ruby-codegen-test-utils/middleware/test_middleware.rb

```diff
@@ -15,4 +15,48 @@ def call(input, context)
       output
     end
   end
+
+  # Middleware used to test relative middleware ordering -
+  # this is for 'BEFORE' type testing
+  class BeforeMiddleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(input, context)
+      context.metadata[:before_middleware] = true
```

Comment:
I believe you can write Rspec expectations here that check if `@app` contains mid and after middleware. And in mid middleware you can check for after and not before, etc. Then you don't need any of the Config values or to check context metadata. You simply just execute the operation and it would fail if it were wrong.

Reply:
Another (possibly better) option is to make these middleware no-op (just @app.call). And in the test, do something like `expect_any_instance_of(MidMiddleware).to receive(:call).ordered` (use ordered to make sure the middleware is called in an order you expect).

Reply:
Updated the test setups so we can expect on output's metadata. (Yay! and thank you for the ideas)